posthog-react-native/src/surveys/components/QuestionTypes.tsx

```diff
@@ -253,7 +253,13 @@ export function MultipleChoiceQuestion({
         appearance={appearance}
         onSubmit={() => {
           // If open choice is selected, replace the choice name with the actual value entered
-          onSubmit(selectedChoices.map((c) => (c === openChoice ? openEndedInput : c)))
+          let result: string | string[] = selectedChoices.map((c) => (c === openChoice ? openEndedInput : c))
+
+          if (result.length === 1) {
+            result = result[0]
+          }
+
+          onSubmit(result)
```

Comment:
```suggestion
          const result = selectedChoices.map(c => c === openChoice ? openEndedInput : c)
          onSubmit(result.length === 1 ? result[0] : result)
```

nit suggestion, feel like in JS we tend to use more functional-style code

this is prob fine though so feel free to ignore